nixos/modules/services/networking/gns3-server.nix

```diff
@@ -0,0 +1,267 @@
+
+    security.wrappers.ubridge = lib.mkIf cfg.ubridge.enable {
+      capabilities = "cap_net_raw,cap_net_admin=eip";
+      group = "ubridge";
+      owner = "root";
+      permissions = "u=rwx,g=rx,o=r";
+      source = "${cfg.ubridge.package}/bin/ubridge";
+    };
+
+    users.users.gns3 = {
+      name = "gns3";
+      group = "gns3";
+      description = "GNS3 user";
+      isSystemUser = true;
+      home = "/var/lib/gns3";
+      extraGroups = lib.optional flags.enableDocker "docker"
+        ++ lib.optional flags.enableLibvirtd "libvirtd"
+        ++ lib.optional cfg.ubridge.enable "ubridge";
+    };
+
+    users.groups.gns3 = { };
+
+    services.gns3-server.settings = lib.mkMerge [
+      {
+        Server = {
+          appliances_path = "/var/lib/gns3/appliances";
+          configs_path = "/var/lib/gns3/configs";
+          images_path = "/var/lib/gns3/images";
+          projects_path = "/var/lib/gns3/projects";
+          symbols_path = "/var/lib/gns3/symbols";
+
+          ubridge_path = lib.mkIf (cfg.ubridge.enable) "${cfg.ubridge.package}/bin/ubridge";
+
+          auth = if cfg.auth.enable then "True" else "False";
+          user = lib.mkIf (cfg.auth.user != null) cfg.auth.user;
+          password = lib.mkIf (cfg.auth.passwordFile != null) "@AUTH_PASSWORD@";
+        };
+      }
+      (lib.mkIf (cfg.vpcs.enable) {
+        VPCS.vpcs_path = lib.mkIf (cfg.vpcs.enable) "${cfg.vpcs.package}/bin/vpcs";
+      })
+      (lib.mkIf (cfg.dynamips.enable) {
+        Dynamips.dynamips_path = lib.mkIf (cfg.dynamips.enable) "${cfg.dynamips.package}/bin/dynamips";
+      })
+    ];
+
+    systemd.services.gns3-server = let
+      commandArgs = "--config /etc/gns3/gns3_server.conf"
+        + " --host ${cfg.host}"
+        + " --pid /run/gns3/server.pid"
+        + " --port ${toString cfg.port}"
+        + lib.optionalString (cfg.log.file != null) " --log ${cfg.log.file}"
+        + lib.optionalString (cfg.log.debug) " --debug"
+        + lib.optionalString (cfg.tls.enable) " --ssl"
+        + lib.optionalString (cfg.tls.certFile != null) " --certfile ${cfg.tls.certFile}"
+        + lib.optionalString (cfg.tls.keyFile != null) " --certkey ${cfg.tls.keyFile}"
+      ;
+    in
+    {
+      description = "GNS3 Server";
+
+      after = [ "network.target" "network-online.target" ];
+      wantedBy = [ "multi-user.target" ];
+      wants = [ "network-online.target" ];
+
+      preStart = ''
+        install -m660 ${configFile} /etc/gns3/gns3_server.conf
+
+        ${lib.optionalString (cfg.auth.passwordFile != null) ''
+          ${pkgs.replace-secret}/bin/replace-secret \
+            '@AUTH_PASSWORD@' \
+            ${cfg.auth.passwordFile} \
+            /etc/gns3/gns3_server.conf
+        ''}
+      '';
+
+      path = lib.optional (flags.enableLibvirtd) pkgs.qemu;
+
+      reloadTriggers = [ configFile ];
+
+      serviceConfig = {
+        ConfigurationDirectory = "gns3";
+        ConfigurationDirectoryMode = "0750";
```

Comment:
Yes, unfortunately I need to store the gns3_server.conf configuration file into a persistent directory like /etc/gns3, because GNS3 Server uses the `--config` base path provided through as CLI argument to store others configuration files at runtime (appliances and the gns3_controller.conf configuration file).

That's why I used ConfigurationMode here, and why I cannot store gns3_server.conf in /run/gns3